MRN-Code/coinstac
# Local Results, Schema Changes and IPCRenderer Fix

* **Problem:**
   1. Local runs not included in results displays.
   1. All controllers `local`.
   1. Closes #372 & closes #369
* **Solution:**
   1. Group GraphQL `remoteRuns` and IndexDB `localRuns` into Redux `runs`.
   1. Controller types now dependent upon presence of *remote* section in computation schema.
   1. Remove auto-inclusion of `covariates` field in pipeline `inputMap`.
   1. All inputs have an owner defined source field.
   1. All inputs, including nested covars, will auto-remap when possible on step position changes.
   1. Collection `fileGroups` made into object to make matching pipeline step output easier.
   1. Empty arrays added to `associatedCollection` to make mapping file output easier.
* **Tests:**
   1. Login and refresh UI.
   1. Pull down computation (remove one first if all already downloaded).
   1. Download complete notification should only display once.
   1. Make **Local Pipeline** active in *Test Consortia* and run it.
   1. Remote server should never be called and results should still be listed in results page.
   1. Other changes should not change functionality.
